cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/deploy/ProgramRunDispatcherFactory.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app.deploy;
+
+import io.cdap.cdap.app.deploy.ProgramRunDispatcher;
+
+public interface ProgramRunDispatcherFactory {
```

Comment:
Why do we need a factory instead of just binding the ProgramRunDispatcher to the right implementation?

Reply:
Because there'll be another implementation of ProgramRunDispatcher ie RemoteProgramRunDispatcher and we'll be deciding which implementation to use based on config, similar to the way it's done for RBAC(Configurator logic).

Reply:
Have removed the factory logic and we're now using a module to bind ProgramRunDispatcher to InMemoryProgramRunDispatcher.